source/particle/world.cc

```diff
@@ -620,6 +625,55 @@ namespace aspect
         property_manager->initialize_one_particle(it);
     }
 
+
+
+#if DEAL_II_VERSION_GTE(9,3,0)
+    template <int dim>
+    void
+    World<dim>::local_update_particles(const typename DoFHandler<dim>::active_cell_iterator &cell,
+                                       const typename ParticleHandler<dim>::particle_iterator &begin_particle,
+                                       const typename ParticleHandler<dim>::particle_iterator &end_particle,
+                                       internal::SolutionEvaluators<dim> &evaluators)
+    {
+      const unsigned int n_particles_in_cell = std::distance(begin_particle,end_particle);
+
+      std::vector<Point<dim> > positions;
+      positions.reserve(n_particles_in_cell);
+
+      for (auto particle = begin_particle; particle!=end_particle; ++particle)
```

Comment:
Once we require deal.II 10.0 I can make this much nicer, by just handing over a particle_iterator_range, but the necessary function in 9.3 is too slow. So yes, this will happen.

Reply:
:-)